File: responser-backend/src/main/java/com/responser/backend/controller/payload/Pagination.java
package com.responser.backend.controller.payload;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

/**
 * Data pagination criteria.
 */
@Data
public class Pagination {

    @NotNull
    @Min(0)
    @Max(Integer.MAX_VALUE)
    private Integer pageNumber;

    @NotNull
    @Min(0)
    @Max(20)
    private Integer pageSize;
}

File: reviewly-chrome-extension/src/app/model/Pagination.ts
export class Pagination {

    public static readonly SINGLE_ELEMENT = new Pagination(0, 1);

    pageNumber: number;

    pageSize: number;

    constructor(pageNumber: number, pageSize: number) {
        this.pageNumber = pageNumber;
        this.pageSize = pageSize;
    }

    public static of = (pageNumber: number, pageSize: number): Pagination => {
        return new Pagination(pageNumber, pageSize)
    }
}
